### PR TITLE
CLOUDP-236434: Fix role mapping patch in FedAuth

### PR DESCRIPTION
**Pull request by helderjs**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

### CODE REVIEW

File: test/int/federated_auth_test.go [hunk 1]
```diff
@@ -19,7 +19,7 @@ import (
 	"github.com/mongodb/mongodb-atlas-kubernetes/v2/test/helper/resources"
 )
 
-var _ = Describe("AtlasFederatedAuth test", Label("AtlasFederatedAuth", "federated-auth"), func() {
+var _ = FDescribe("AtlasFederatedAuth test", Label("AtlasFederatedAuth", "federated-auth"), func() {
```
Comment (igor-karpukhin) [inline test/int/federated_auth_test.go:22]: Looks like a left-over from the local testing :)

File: pkg/api/v1/atlasfederatedauth_types.go [hunk 2]
```diff
@@ -67,7 +67,6 @@ func (f *AtlasFederatedAuthSpec) ToAtlas(orgID, idpID string, projectNameToID ma
 		}
 		atlasRoleMappings = append(atlasRoleMappings, admin.AuthFederationRoleMapping{
 			ExternalGroupName: roleMapping.ExternalGroupName,
-			Id:                &idpID,
```
Comment (igor-karpukhin) [inline pkg/api/v1/atlasfederatedauth_types.go:70]: Does it work without providing the ID?

Comment (helderjs) [reply pkg/api/v1/atlasfederatedauth_types.go:70]: Yes

File: test/int/federated_auth_test.go (hunk 1 above)
Comment (helderjs) [reply test/int/federated_auth_test.go]: We need a rule to catch this in the linter :grimacing:

**Review by igor-karpukhin — APPROVED** [review]
Great fix 👍

**Review by josvazg — APPROVED** [review]

**Pull request merged**